### PR TITLE
[WIP] Update java checker

### PR DESCRIPTION
**Pull request by wsdjeg**
TODO:
- [x] auto generate classpath
- [x] auto detect output directry
- [x] maven support
- [x] gradle support
- [x] support eclipse project's `.classpath` file, for this feature,you need python support.
- [x] support checkstyle

### CODE REVIEW

File: autoload/neomake/makers/ft/java.vim [hunk 1]
```diff
@@ -1,10 +1,57 @@
...
+            \ '%W%f:%l: warning: %m,'.
+            \ '%E%f:%l: %m,'.
+            \ '%Z%p^,'.
+            \ '%-G%.%#'
          \ }
 endfunction
+
+function! neomake#makers#ft#java#checkstyle()
+    return {
+        \ 'args': ['-c', '/usr/share/checkstyle/google_checks.xml'],
```
Comment (wsdjeg) [inline autoload/neomake/makers/ft/java.vim]: @paretje I have merged your branch,here I think maybe we can provide a g:var for user to set where is the xml file.

File: autoload/neomake/makers/ft/java.vim [hunk 2]
```diff
@@ -16,27 +16,134 @@ set cpo&vim
...
+if s:is_windows
+    let s:fsep = ';'
+    let s:psep = '\'
+else
+    let s:fsep = ':'
+    let s:psep = '/'
+endif
+let g:neomake_java_checker_home = fnamemodify(expand('<sfile>'), ':p:h:gs?\\?/?')
+
+" customer options
```
Comment (brcolow) [inline autoload/neomake/makers/ft/java.vim]: customer => custom

Comment (wsdjeg) [inline autoload/neomake/makers/ft/java.vim]: Thanks 👍

**Comment by wsdjeg** [comment]
@blueyed for this PR,I need import two files into this repo,where should I put the files?the files is just for java maker, should I put it `autoload/neomake/makers/ft/java/build.gradle`?

**Comment by blueyed** [comment]
@wsdjeg
Not sure, but seems good!

**Comment by wsdjeg** [comment]
ok，thanks，I think I will finish this PR tonight.But will not support custom config file.I do not think it is needed for a java project.

**Comment by wsdjeg** [comment]
github's merge squash looks not very good.It can not automerge patch from other people.

**Comment by blueyed** [comment]
> It can not automerge patch from other people.

What do you mean?

**Comment by andjscott** [comment]
I get this error when calling Neomake from a Java file in a gradle project.

    Error detected while processing function neomake#Make[3]..neomake#GetEnabledMakers[23]..neomake#utils#AvailableMakers[1]..neomake#utils#MakerIsAvailable[7]..neomake#GetMaker[27]..neomake#makers#ft#java#javac[22]..<SNR>108_GetGradleClasspath:
    line   29:
    E716: Key not present in Dictionary: build.gradle
    E15: Invalid expression: g:neomake_java_javac_gradle_classpath[gradle]
    Error detected while processing function neomake#Make[3]..neomake#GetEnabledMakers[23]..neomake#utils#AvailableMakers[1]..neomake#utils#MakerIsAvailable:
    line    7:
    E171: Missing :endif
    Error detected while processing function neomake#Make[3]..neomake#GetEnabledMakers:
    line   23:
    E171: Missing :endif
    Error detected while processing function neomake#Make:
    line    3:
    E171: Missing :endif
    Press ENTER or type command to continue

**Comment by wsdjeg** [comment]
sorry，thank for quick fix

**Pull request merged**